### DISCUSSION TITLE
invalidatesTags works even if the request is 422 http status

### DISCUSSION THREAD

**Discussion by phoique** [reduxjs/redux-toolkit#3514] [Q&A] [answered]
Hello, I have a mutation code in the form of the following code. invalidatesTags when request is 200. While waiting for it to work, it works in http statuses such as 422, 409. Is this a problem or am I setting it up wrong?

```
testMutation: builder.mutation({
	query: (body) => ({
		url: 'test/services',
		method: 'POST',
		body,
	}),
	invalidatesTags: ptpApi.invalidatesTags(['testInvalidatesTags']),
	extraOptions: {
		toastSuccessMessageKey: 'testMutation',
		toastErrorMessageKey: 'testMutation',
	},
}),
```

**Comment by EskiMojo14** [comment, marked as answer]
if you provide an array for invalidatesTags, it will invalidate those tags for both successful and unsuccessful requests.

if you only want to invalidate tags for successful requests, you should provide a callback:
```ts
testMutation: builder.mutation({
	query: (body) => ({
		url: 'test/services',
		method: 'POST',
		body,
	}),
	invalidatesTags: (result, error, arg) => !error ? ptpApi.invalidatesTags(['testInvalidatesTags']) : [],
	extraOptions: {
		toastSuccessMessageKey: 'testMutation',
		toastErrorMessageKey: 'testMutation',
	},
}),
```

**Reply by phoique** [reply]
can i write it as middleware i used it as i wrote in all my project.

**Reply by EskiMojo14** [reply]
i don't understand the question, sorry

**Reply by phoique** [reply]
There are close to 100 endpoints in my project. I used invalidatesTags as I said. Isn't there an easy way?

**Reply by EskiMojo14** [reply]
using a callback, like i said?
